src/pages/Dashboard.tsx

```diff
@@ -81,10 +102,37 @@ const Dashboard = () => {
         }
     }
 
+    const createNewGroup = async (groupName: string) => {
+        try {
+            await addUserToGroup(TEST_USER, groupName);
+            setUserGroups(currentUserGroups.concat(groupName));
+            await switchViewedGroup(groupName);
+            setNewGroupNameInput("");
+        } catch (error: unknown) {
+            // Replace with user visible messaging
+            console.log("Could not create new group");
+        }
+    }
+
+    const leaveGroup = async (groupName: string) => {
+        try {
+            await removeUserFromGroup(TEST_USER, groupName);
+            const filteredUserGroups: string[] = currentUserGroups.filter((userGroup) => userGroup !== groupName);
+            setUserGroups(filteredUserGroups);
```

Comment:
You can combine these into one step by using `setUserGroups(prevFilteredUserGroups => prevFilteredUserGroups.filter(...))`

Reply:
The dedicated reference to the filtered user groups is just so the new user groups can be accessed a bit later in the function

Reply:
You can use map then and put the other setState in the map but we can do it later